File: 1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minPairSum = function (nums) {
    const n = nums.length;
    let maxsum = 0;
    nums.sort((a, b) => a - b);

    for (let i = 0; i < n / 2; i++) {
        maxsum = Math.max(maxsum, nums[i] + nums[n - i - 1]);
    }

    return maxsum;
};

File: 1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.cpp
class Solution {
public:
    int minPairSum(vector<int>& nums) {
        int n = nums.size();
        sort(nums.begin(), nums.end());

        int maxsum = 0;
        for (int i = 0; i < n / 2; i++) {
            maxsum = max(maxsum, nums[i] + nums[n - 1 - i]);
        }

        return maxsum;
    }
};